compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -541,24 +538,23 @@ static bool isFusableWithProducer(
   auto consumerLinalgOp = cast<linalg::LinalgOp>(consumer);
   if (consumerLinalgOp.isDpsInput(&operand)) {
     // Only fuse on inputs if both ops are generic ops.
-    if (!aggressiveFusion || !isa<linalg::GenericOp>(consumer) ||
+    if (!fuseMultiUse || !isa<linalg::GenericOp>(consumer) ||
```

Comment:
This one too is not related to `fuseMultiUse`... Can we try making this 
```
if (!isa<linalg::GenericOp>(consumer) || !isa<linalg::GenericOp>(producer)) {
  return false;
}
```

and see if there are any issues....

Reply:
Landed in #12589